tox.ini

```diff
@@ -0,0 +1,84 @@
+[tox]
+skipsdist=True
+skip_missing_interpreters = True
+envlist = fmt, lint, unit
+
+[vars]
+src_path = {toxinidir}/src/
+tst_path = {toxinidir}/tests/
+lib_path = {toxinidir}/lib/charms/nrpe/
+all_path = {[vars]src_path} {[vars]tst_path}
+
+[testenv]
+setenv =
+  PYTHONPATH = {toxinidir}:{toxinidir}/lib:{[vars]src_path}
+  PYTHONBREAKPOINT=pdb.set_trace
+  PY_COLORS=1
+  juju2: LIBJUJU="2.9.44"  # libjuju2
+  juju3: LIBJUJU="3.2.2"  # libjuju3
+passenv =
+  PYTHONPATH
+  CHARM_BUILD_DIR
+  MODEL_SETTINGS
+
+[testenv:fmt]
+description = Apply coding style standards to code
+deps =
+    # renovate: datasource=pypj
+    black==23.7.0
+    # renovate: datasource=pypi
+    ruff==0.0.287
+commands =
+    ruff --fix {[vars]all_path}
+    black {[vars]all_path}
+
+[testenv:lint]
+description = Check code against coding style standards
+deps =
+    # renovate: datasource=pypi
+    black==23.7.0
+    # renovate: datasource=pypi
+    ruff==0.0.287
+    # renovate: datasource=pypi
+    codespell==2.2.5
+commands =
+    codespell {toxinidir}
+    ruff {[vars]all_path}
+    black --check --diff {[vars]all_path}
+
+[testenv:unit]
+description = Run unit tests
+deps =
+    -r{toxinidir}/requirements.txt
+    # renovate: datasource=pypi
+    pytest==7.4.1
+    # renovate: datasource=pypi
+    coverage[toml]==6.5.0
+    # renovate: datasource=pypi
+    pydantic <= 2.0
```

Comment:
I couldn't find any place where we use pydantic in this repo:
https://github.com/search?q=repo%3Ajat-canonical%2Faar-operator+pydantic&type=code
turns up with nothing (except `tox.ini`). Am I missing something?